src/web/app/pages-instructor/instructor-session-individual-extension-page/instructor-session-individual-extension-page.component.ts

```diff
@@ -0,0 +1,544 @@
+  SortBy: typeof SortBy = SortBy;
+  SortOrder: typeof SortOrder = SortOrder;
+  sortStudentsBy: SortBy = SortBy.SESSION_END_DATE;
+  sortStudentOrder: SortOrder = SortOrder.DESC;
+  sortInstructorsBy: SortBy = SortBy.SESSION_END_DATE;
+  sortInstructorOrder: SortOrder = SortOrder.DESC;
+
+  isAllStudentsSelected: boolean = false;
+  isAllInstructorsSelected: boolean = false;
+
+  courseId: string = '';
+  courseName: string = '';
+  feedbackSessionName: string = '';
+
+  feedbackSessionEndingTime: number = 0;
+  feedbackSessionTimeZone: string = 'UTC';
+
+  feedbackSessionDetails: FeedbackSessionBasicRequest = {
+    instructions: '',
+    submissionStartTimestamp: 0,
+    submissionEndTimestamp: 0,
+    gracePeriod: 0,
+    sessionVisibleSetting: SessionVisibleSetting.AT_OPEN,
+    customSessionVisibleTimestamp: 0,
+    responseVisibleSetting: ResponseVisibleSetting.CUSTOM,
+    customResponseVisibleTimestamp: 0,
+    isClosingEmailEnabled: false,
+    isPublishedEmailEnabled: false,
+  };
+
+  columnsData: ColumnData[] = [];
+  rowsData: SortableTableCellData[][] = [];
+
+  studentsOfCourse: StudentExtensionTableColumnModel[] = [];
+  instructorsOfCourse: InstructorExtensionTableColumnModel[] = [];
+  studentDeadlines: Record<string, number> = {};
+  instructorDeadlines: Record<string, number> = {};
+
+  isLoadingAllStudents: boolean = true;
+  hasLoadedAllStudentsFailed: boolean = false;
+  isLoadingAllInstructors: boolean = true;
+  hasLoadedAllInstructorsFailed: boolean = false;
+  isLoadingFeedbackSession: boolean = true;
+  hasLoadingFeedbackSessionFailed: boolean = false;
+  isSubmittingDeadlines: boolean = false;
+
+  ngOnInit(): void {
+    this.route.queryParams.subscribe((queryParams: any) => {
+      this.courseId = queryParams.courseid;
+      this.feedbackSessionName = queryParams.fsname;
+      this.loadFeedbackSessionAndIndividuals();
+    });
+  }
+
+  constructor(
+    private statusMessageService: StatusMessageService,
+    private feedbackSessionsService: FeedbackSessionsService,
+    private studentService: StudentService,
+    private ngbModal: NgbModal,
+    private route: ActivatedRoute,
+    private courseService: CourseService,
+    private tableComparatorService: TableComparatorService,
+    private instructorService: InstructorService,
+  ) {}
+
+  /**
+   * Loads a feedback session and individuals
+   */
+  loadFeedbackSessionAndIndividuals(): void {
+    this.resetTables();
+    this.isLoadingAllStudents = true;
+    this.hasLoadedAllStudentsFailed = false;
+    this.isLoadingFeedbackSession = true;
+    this.hasLoadingFeedbackSessionFailed = false;
+    this.isLoadingAllInstructors = true;
+    this.hasLoadedAllStudentsFailed = false;
+    forkJoin([
+      this.courseService.getCourseAsInstructor(this.courseId),
+      this.feedbackSessionsService.getFeedbackSession({
+        courseId: this.courseId,
+        feedbackSessionName: this.feedbackSessionName,
+        intent: Intent.FULL_DETAIL,
+      }),
+    ])
```

Comment:
just a sidenote! I'm good with this as it is but it could lead to indeterminacy in error messages - for e.g. if a course ID is incorrect, the error message could either be from getting the course or the feedback session

Reply:
That is true too. I believe for the majority of use cases, this should be provide a better user experience, so I'll leave it here for now

Reply:
Try not to use forkJoin which will create burst of network calls. Only use it when you can justify why it must be added:)

Reply:
Would it be ok if there's just 2 calls in this case? though yeah not sure if using forkJoin here would be that idiomatic - I'm kinda alright with either approach (also for ref https://github.com/TEAMMATES/teammates/pull/11603#discussion_r835206407)

Reply:
Personally don't see any issues with this, considering forkJoin is used for many similar calls in the frontend.

Reply:
It is ok then. The thing about forkJoin is that we will most likely have 2 servlet for the same request (instead of 1) and the Google API gateway would spin up extra instances if there are too many concurrent requests. I personally think the use of it should be justified (a way better UX).